### PR TITLE
Change the way we instantiate resources. 

### PR DESCRIPTION
**Pull request by miogalang**
Create new instances of each resource when calling them. 
`$client->tickets()->comments()` would instantiate new classes for `Tickets` and `Comments`. 
This means we do not have to worry about leftover state variables and is better design overall.

Additionally we want to add consistency and always use plural both for the resources class names and when instantiating them.

/cc @joseconsador @jwswj @mmolina @atroche 

### References
 - Jira link: 

### Risks
 - We might not be able to chain properly (medium)

### CODE REVIEW

File: src/Zendesk/API/HttpClient.php [hunk 1]
```diff
@@ -16,41 +19,8 @@
  * @method Debug debug()
  * @method Tickets ticket()
  * @method Tickets tickets()
- * @method TicketFields ticketFields()
```
Comment (miogalang) [inline src/Zendesk/API/HttpClient.php]: Removed all the stuff unavailable in V2. Add these as you add resources.

**Comment by joseconsador** [comment]
:+1:

**Comment by atroche** [comment]
:+1:

**Comment by jwswj** [comment]
:+1:

**Pull request merged**